tools/release.sh

```diff
@@ -42,27 +42,28 @@ shift $((OPTIND-1))
 echo "# Selecting GPG key ..."
 
 gpgkey=$(gpg --list-secret-keys --keyid-format SHORT | awk -F'( +|/)' '/^(sec|ssb)/{print $3}')
-keycount=$(echo $gpgkey | wc -w)
+keycount=$(echo "$gpgkey" | wc -w)
 
-if [ $keycount -eq 0 ]; then
+if [ "$keycount" -eq 0 ]; then
+  # shellcheck disable=SC2016
   echo 'Need at least one GPG key, please make one with `gpg --gen-key`'
   echo 'You will also need to submit your key to a public keyserver, e.g.'
   echo '  https://sks-keyservers.net/i/#submit'
   exit 1
-elif [ $keycount -ne 1 ]; then
-  echo -e 'You have multiple GPG keys:\n'
+elif [ "$keycount" -ne 1 ]; then
+  printf "You have multiple GPG keys:\n\n"
 
   gpg --list-secret-keys
 
   while true; do
-    echo $gpgkey | awk '{ for(i = 1; i <= NF; i++) { print i ") " $i; } }'
-    echo -n 'Select a key: '
-    read keynum
-
-    if $(test "$keynum" -eq "$keynum" > /dev/null 2>&1); then
-      _gpgkey=$(echo $gpgkey | awk '{ print $'${keynum}'}')
-      keycount=$(echo $_gpgkey | wc -w)
-      if [ $keycount -eq 1 ]; then
+    echo "$gpgkey" | awk '{ for(i = 1; i <= NF; i++) { print i ") " $i; } }'
+    printf 'Select a key: '
+    read -r keynum
+
+    if [ "$keynum" -eq "$keynum" ]; then
```

Comment:
Other nit - change wise it looks fine but why is this script actually checking `"$keynum" -eq "$keynum"`?

Reply:
Is this a really complicated way to check if keynum is an integer?

Reply:
apparently so, ugly https://stackoverflow.com/a/29852684

Reply:
Then do we need to restore the `> /dev/null 2>&1` bit?

Reply:
Actually can this just be 
```suggestion
    if [ -z "${keynum##*[!0-9]*}" ]; then
```

Reply:
I think it is, a quick Google search reveals the `test` command implemented before in [this StackOverflow answer](https://stackoverflow.com/questions/3623662/testing-if-a-variable-is-an-integer).

Reply:
You got to love how [the answer suggesting this](https://stackoverflow.com/a/3626205) is spawning an extra subshell and returning stdout (which it doesn't care about, `if`s just look at exit status) for the lols.

Like:

```sh
if $(is_int "${input}");
```
Is a super weird way to do:
```sh
if is_int "$input"
```

I'll actually go ahead and fix it for good measure 🙃

Reply:
And an extra integer expression - but with the old deprecated `$[]` syntax for the lols in the echo :D

Reply:
I've refactored the infinite loop, PTAL